plugins/rapture/nexus-rapture-plugin/src/main/java/org/sonatype/nexus/rapture/internal/security/SessionAuthenticationFilter.java

```diff
@@ -52,6 +52,19 @@
 
   public static final String P_REMEMBER_ME = "rememberMe";
 
+  public static final String DELETE_METHOD = "DELETE";
+
+  /**
+   * Allow access, by default rules, and also if this is a request to logout.
+   */
+  @Override
+  protected boolean isAccessAllowed(final ServletRequest request,
+                                    final ServletResponse response,
+                                    final Object mappedValue)
+  {
+    return super.isAccessAllowed(request, response, mappedValue) || isLogoutRequest(request, response);
```

Comment:
Just suggesting reversing order of operations in the OR condition and checking isLogoutRequest first, with the assumption that isAccessAllowed() is more expensive to check